server/controllers/ArticleController.js

```diff
@@ -0,0 +1,201 @@
+ */
+class ArticleController {
+  /**
+   * Method adds a new article to the database
+   * It returns a json response of the article created
+   * It handles the POST /api/articles.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static createArticle(req, res, next) {
+    const {
+      title, description, body,
+    } = req.body.article;
+
+      // generating a slug
+    const id = uuid();
+    const slug = slugCreate(`${title.toLowerCase()}-${id}`);
+    const authorId = req.body.decoded.id;
+
+
+    const {
+      username, bio, image,
+    } = req.user;
+
+    Article.create({
+      id, slug, title, description, body, authorId,
+    })
+      .then(article => res.status(201).json({
+        article: articleControllerHelper
+          .createArticleResponse(article, username, bio, image, slug),
+        message: 'Article Sucessfully Created',
+      })).catch(next);
+  }
+
+  /**
+   * Method gets an article from the database,
+   * It returns a json response of the article retrieved
+   * It handles the GET /api/articles/:slug.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static getArticle(req, res, next) {
+    Article.find({
+      where: { slug: req.params.slug },
+      include: [{
+        model: User,
+        as: 'author',
+        attributes: [
+          'username', 'bio', 'image',
+        ],
+      }],
+      attributes: [
+        'slug', 'title', 'description',
+        'body', 'createdAt', 'updatedAt',
+      ],
+    })
+      .then((article) => {
+        if (!article) {
+          return res.status(404).json({
+            status: 'fail',
+            message: 'Article not found',
+          });
+        }
+        return res.status(200).json({
+          article,
+          message: 'Article Found',
+        });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method gets all articles from the database,
+   * It returns a json response of the article retrieved
+   * It handles the GET /api/articles.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static getAllArticle(req, res, next) {
+    Article.findAndCountAll({
+      include: [{
+        model: User,
+        as: 'author',
+        attributes: [
+          'username', 'bio', 'image',
+        ],
+      }],
+      attributes: [
+        'slug', 'title', 'description',
+        'body', 'createdAt', 'updatedAt',
+      ],
+      order: [['createdAt', 'DESC']],
+    })
+      .then((articles) => {
+        if (!articles.count) {
+          return res.status(404).json({
+            status: 'fail',
```

Comment:
You always use `status: 'fail'` for your errors, I don't think it is necessary, coupled with the fact that it has the same `key` as the status code: `status`.

Reply:
done. It has been removed